jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -434,32 +434,71 @@ public boolean remove(Connection connection)
         if (removed)
         {
             released(connection);
-            removed(connection);
+            onRemoved(connection);
         }
         return removed;
     }
 
+    /**
+     * <p>Callback method invoked when a new {@link Connection} has been created.</p>
+     *
+     * @param connection the {@link Connection} that has been created
+     * @see #onRemoved(Connection)
+     */
     protected void onCreated(Connection connection)
     {
     }
 
+    /**
+     * @param connection the {@link Connection} that become idle
+     * @param close whether this pool is closing
+     * @return {@code true} to indicate that the connection is idle, {@code false} otherwise
+     * @deprecated do not override because its usage is racy
```

Comment:
This text is vague.
Is the API just racy?
Can it be used?
Is there a replacement?

This is a form of limp-wristed writing, Use something more direct instead.

> Racy API.  Do not use.  There is no replacement.